### ISSUE TITLE
Pay Quadratic Rewards to early UNI-LPs

### ISSUE THREAD

**Issue by ghost** [0xHabitat/improvements-and-bugs#4]
## Summary
Paying out HBT Rewards to all LPs which joined the HBT-ETH Pool from Genesis Day, the 10th March 2021.

## Abstract
The objective of this proposal is to incentivise LP allocation in the following HBT-ETH Pair on Uniswap:
> https://info.uniswap.org/pair/0xc7a1cb6edc22e94f17c80eb5b959f2ad28511d4e

Through a liquidity incentive, the reserve can distribute additional voting power to the early members of the Habitat Community, making it the first rewards distribution after the initial launch.

## ⚡️ Motivation // Challenge
Both LPs and early community-members can profit from the early provision of additional capital in Uniswap.

Higher Liquidity means a natural and organic price discovery, invested community members earn fees and an additional HBT Reward, which is distributed fairly from June on.

This additional liquidity helps attracting more Token-holders and extend the reach of Habitat. Further steps can be undertaken, listings on official pages secured and the successful implementation showcases the coordinative strength of our community.

For the Genesis Team, it has been very impressive to see the vocal community that has formed around Habitat. We want to allocate the first 2 batches of unlocked Liquidity Funds to early LPs.

## 🦋  Desired Outcome
* Pay 180.012 HBT (2143 HBT per day) to LPs on Uniswap Pair HBT-ETH pool which provide Liquidity between the 10th March 2021 and 02nd June 2021 // Total of 84 days.
* After 09th June Liquidity Funds are unlocked and ready to be dripped linearly to all members, as stated in this distribution table: http://bit.ly/HBT-distribution

The liquidity provided is going to be averaged out over 84 days. LPs only need to hold the UNI-LP Token, **no locking or staking is needed**.

This keeps flexibility for LPs, saves the gas costs of interacting with the contract and protects holders from bugs or hacks. Using the [droplet contracts](https://github.com/0xHabitat/droplet) helps to ensure the process works without calling `claim` on a staking contract and using gas to do so.

*For example:*
If you put liquidity in on Day 1 and take some part of it out on Day 6, we calculate the mean value of liquidity provided across the full 84 days.

## 🎯 Action Points
* The Habitat team will setup a mechanism to read out `Mint & Burn` events, meaning deposit times and deposit durations of LP Holders of HBT-ETH.
* Rewards are distributed in a fair manner via [droplet contracts](https://github.com/0xHabitat/droplet) after the bootstrap phase ends in beginning of June.

**Reward Algorithm**
> https://github.com/0xHabitat/habitat/blob/master/web/landing/lp/index.js
```
const ONE_WEEK = 3600 * 24 * 7; // in seconds
const MAX_WEEKS = ONE_WEEK * 12; // max. weeks
const MAX_HBT = 2_000_000; // current circulating supply
const MAX_SHARE = Math.sqrt(MAX_HBT) * (MAX_WEEKS * MAX_WEEKS);
const MAX_REWARD = 2143; // (per day) * 84 = 180.012
const DAY = 3600 * 24; // one day in seconds

function calcDailyReward (amount) {
  let cumulative = 0;
  const rewards = [];

  for (let seconds = 0; seconds < MAX_WEEKS; seconds += DAY) {
    const share = (Math.sqrt(amount) * (seconds * seconds)) / MAX_SHARE;
    const r = MAX_REWARD * share;
    rewards.push(r);
    cumulative += r;
  }

  return { cumulative, rewards };
}

```

#### We can achieve:
* Increase HBT Liquidity on Uniswap.
* Creating more stakeholders for HBT Community.
* Extend Habitat's reach across the ecosystem.

#### To be discussed:
* over which period of time the rewards will be streamed to LPs after 09th June 2021?

## 🐇 Additional Information
**TL;DR:**
* 180k HBT will be distributed to HBT-ETH LP providers that leave their capital in the [Uniswap Pair](https://info.uniswap.org/pair/0xc7a1cb6edc22e94f17c80eb5b959f2ad28511d4e)
* The active period for the reward starts on the Genesis Day 10.03.2021, and will end 84 days later on 2nd June 2021.
* After the period has concluded, Habitat will use the in-house Droplet Contracts to distribute the tokens to all eligible addresses, starting from 09th June 2021.
* A post-mortem of the process will show the impacts of this proposal and start a discussion regarding possible repetitions in the future.

**Comment by ghost** [comment]
**📣 Update**

[Quadratic Reward Calculator](https://0xhabitat.org/lp/) - enter your HBT amount and check how many rewards you would get after 02nd June, if you would deposit right now.
